torch/quantization/fx/quantize.py

```diff
@@ -549,11 +553,18 @@ def load_arg(a: Argument) -> Argument:
 
 
         if all_node_args_have_no_tensors(node, modules, cache_for_no_tensor_check):
-            non_tensor_input_binary_op_nodes.add(node)
+            non_tensor_input_nodes.add(node)
         if root_node is None and node.op != 'placeholder':
+            # rule 4: remove observer for getitem if it is followed by an unmatched node
+            observer_node = node.args[0]
+            if len(observer_node.args) > 0:
+                observed_node = observer_node.args[0]
+                if is_activation_post_process_node(observer_node, modules) and \
+                   (observed_node.op, observed_node.target) == ("call_function", operator.getitem):
+                    actpp_to_remove.add(observer_node)
             unmatched_nodes.add(node)
 
-        # rule 3: for special node, we'll just remove observer for its input
+        # rule 5: for special node, we'll just remove observer for its input
```

Comment:
`getitem` is also handled here, would it make sense to combine the two?

Reply:
this one is removing the observer for input of getitem actually. currently the handling of getitem is a bit messy because we only want to observe the getitem nodes whose output is a Tensor, and this information is not present currently in a GraphModule, we can simplify the handling if we know the type of a Proxy.